framework/src/controller/application.js

```diff
@@ -232,26 +225,32 @@ class Application {
 	 * @param {number} transactionType - Unique integer that identifies the transaction type
 	 * @param {constructor} Transaction - Implementation of @liskhq/lisk-transactions/base_transaction
 	 */
-	registerTransaction(transactionType, Transaction, options = {}) {
-		// TODO: Validate the transaction is properly inherited from base class
+	registerTransaction(Transaction, matcher) {
```

Comment:
To make sure future improvements may not break the API. 

```suggestion
	registerTransaction(Transaction, { matcher }) {
```